### DISCUSSION TITLE
Why "VueButton" is considered a bad component name and "VButton" is not?

### DISCUSSION THREAD

**Discussion by lanaschuster** [vuejs/docs#2510] [Q&A] [answered]
"Why is "VueButton" considered a bad component name while "VButton" is not? The "Base component names" (B Rules) are not clear to me.

In the detailed explanation, it says: "Since component names should always be multi-word, this convention prevents you from having to choose an arbitrary prefix for simple component wrappers (e.g., MyButton, VueButton)."

Why a single letter as a prefix is not considered arbitrary?

Style guide link: https://vuejs.org/style-guide/rules-strongly-recommended.html#base-component-names

**Comment by brc-dd** [comment, marked as answer]
Technically single letter prefix can be considered arbitrary too. But mostly people use them to avoid naming conflicts or to better organize stuff. This is especially common with component libraries. Like vuetify uses `v` as prefix, element plus uses `el`, ant design uses `a`, and so on. And in projects too (suppose it's named `Foo`), and [you won't wanna have just `Button`](https://vuejs.org/style-guide/rules-essential.html#use-multi-word-component-names), so you will need to make every component have at least two words, and people often go with `FButton` (F is the first character of Foo; people don't usually go with `FooButton` as that's just longer to type).

**Comment by canyon920** [comment]
single-word component is Inconvenience. 
Javascript language is better word is Camel case
this is you can component customing for best
